docs/markdown/Contributions/development/style-guide.md

```diff
@@ -389,6 +389,41 @@ merge_constraints([constraint for constraint in all_constraints if constraint.st
 
 The return type, however, should usually be as precise as possible so that call sites have better type inference.
 
+### Use descriptive names for `TypeVar`s
+
+#### Bounded `TypeVar`s
+
+Bounded `TypeVar`s should be named after their `bound` type, followed by a single letter (starting with `T`, continuing on to `U`, `V`, etc if multiple `TypeVar`s with the same bound are required.):
+
+```python
+_ListElementT = TypeVar("_ListElementT", bound=ListElement)
+```
+
+#### Unbounded `TypeVar`s
+
+Unbounded `TypeVars` should have a descriptive name, for example in `selectors.py`, output types for `Get` and `MultiGet` are named `_Output`, `_Out1`, `_Out2`, etc.
```

Comment:
To me it looks counter intuitive to draw examples from `selectors.py` using type vars that does not follow the recommended style of ending with `T`.

Reply:
... or is that only for bounded type vars? 

I would prefer if we had the same style for all type vars, only that bounded ones use the bounded type in its name, which I think is a good idea.

Reply:
I think I'm +1 on that as well. It helps the brain make a mental leap that this _thing_ is a typevar, not a real type.

Reply:
that, and I'm onboard also with @cognifloyd's suggestion with sticking with `T` then numerals if more of the same is required.